server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -306,60 +287,49 @@ private void registerRequestHandler(DynamicActionRegistry dynamicActionRegistry)
-            } catch (IOException e) {
-                throw new IOException("Could not read from extensions.yml", e);
-            }
-            for (Extension extension : extensions) {
-                loadExtension(extension);
-            }
-            if (!extensionIdMap.isEmpty()) {
-                logger.info("Loaded all extensions");
-            }
-        } else {
-            logger.warn("Extensions.yml file is not present.  No extensions will be loaded.");
+    }
+
+    private void validateField(String fieldName, String value) throws IOException {
+        if (Strings.isNullOrEmpty(value)) {
+            throw new IOException("Required field [" + fieldName + "] is missing in the request");
         }
     }
 
-    /**
-     * Loads a single extension
-     * @param extension The extension to be loaded
-     */
-    private void loadExtension(Extension extension) throws IOException {
+    private void validateExtension(Extension extension) throws IOException {
+        validateField("extension name", extension.getName());
+        validateField("extension uniqueId", extension.getUniqueId());
+        validateField("extension host address", extension.getHostAddress());
+        validateField("extension port", extension.getPort());
+        validateField("extension version", extension.getVersion());
+        validateField("opensearch version", extension.getOpensearchVersion());
+        validateField("minimum opensearch version", extension.getMinimumCompatibleVersion());
         if (extensionIdMap.containsKey(extension.getUniqueId())) {
-            logger.info("Duplicate uniqueId " + extension.getUniqueId() + ". Did not load extension: " + extension);
-        } else {
-            try {
-                DiscoveryExtensionNode discoveryExtensionNode = new DiscoveryExtensionNode(
-                    extension.getName(),
-                    extension.getUniqueId(),
-                    new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
-                    new HashMap<String, String>(),
-                    Version.fromString(extension.getOpensearchVersion()),
-                    Version.fromString(extension.getMinimumCompatibleVersion()),
-                    extension.getDependencies()
-                );
-
-                extensionIdMap.put(extension.getUniqueId(), discoveryExtensionNode);
-                extensionSettingsMap.put(extension.getUniqueId(), extension);
-                logger.info("Loaded extension with uniqueId " + extension.getUniqueId() + ": " + extension);
-            } catch (OpenSearchException e) {
-                logger.error("Could not load extension with uniqueId " + extension.getUniqueId() + " due to " + e);
-            } catch (IllegalArgumentException e) {
-                throw e;
-            }
+            throw new IOException("Duplicate uniqueId " + extension.getUniqueId() + ". Did not load extension: " + extension);
```

Comment:
In other responses we tend to put square brackets around the inserted string, might want those here.